### ISSUE TITLE
Some improvement suggestions and ideas

### ISSUE THREAD

**Issue by FlyingBananaTree** [rashevskyv/dbi#48]
As someone who uses DBI a lot here are my ideas and suggestions for improving the application as it exists at version 314.

Suggestion 1. Allow highlighting the user's language with a config entry for example `LanguageHighlight=JA`, right now only Russian is highlighted:
 
![language](https://user-images.githubusercontent.com/74322834/127054111-bcf0d002-59d1-4208-a975-b009c3285be3.png)

Suggestion 2. Add a method for changing the game name that DBI displays throughout the application and even MTP folders if possible. This would be very useful if you have several Japanese games installed for example as it can be difficult to find what you're looking for.
 
Ideas for implementing this:

- first idea: with https://github.com/p-sam/switch-sys-tweak you can override the game name which is useful for changing it on the homescreen and it uses the same file that DBI writes to for changing atmosphere region and language settings- `atmosphere/contents/titleid/config.ini`

The format is as follows:
```
[override_nacp]
name=
author=
display_version=
```

Therefore have DBI just read the `name=` value from `config.ini`

- second idea: offer a game name section in `dbi.config`

Suggestion 3. Issue https://github.com/rashevskyv/dbi/issues/46 - atmosphere no longer includes the contents folder by default so create the contents folder if it doesn't exist so that DBI can write the config.ini file

Suggestion 4. Issue https://github.com/rashevskyv/dbi/issues/34 - this issue was closed but I agree with this, scrolling through the game list is painfully slow even with the shoulder button shortcuts and scrolling doesn't work with the d-pad.

Suggestion 5. Add the game name to the zip file created by the save game back up feature, this would make it much easier to find a specific save on your computer quickly at a glance and without needing to check the titleid or open the zip and check `.dbi_save_info.ini`

**Comment by duckbill007** [comment]
1. I will think about it.
2. Yes, first idea is good. As for second - I will think about it.
3. Thank you! I forgot about that and thought that contents always exist in atmosphere.
4. Maybe somewhen iin the future. Not now.
5. There is a problem: nintendo's fat driver allows only ascii symbols in filenames. That's why I left only titleID and date. Also this is the reason why ZIP only, no plain files: names in save technically can be any, I saw in some JP game save with JP symbols in filenames.

**Comment by duckbill007** [comment]
2. sys-tweak overrides only for qlaunch process. So, there will be custom config, because scanning contents folder is slow.

**Comment by FlyingBananaTree** [comment]
2. Ok, would have been a nice feature but not worth it if the scan is slow.

5. I forgot to mention that. I use JKSV and the way it names the directory: `Game name/save_name.zip` or if a unicode character is detected it checks `switch/JKSV/titleDefs.txt` for a custom entry such as `01009040091E0000 = "Wolfenstein II - The New Colossus"` then finally if the custom entry isn't there the fall back is `titleid/save_name.zip`

It's not that common that title contains unicode characters so I think the fall back is pretty good even without using `titleDefs.txt`

Maybe the custom config you are talking about can cover suggestion 2 and suggestion 5 at the same time?

The zip files are really nice since they contain all the loose save files in a single file which is better for transferring to computer and storing. Is drag and drop zip to DBI's MTP save directories and have them auto extract possible by the way?

**Comment by duckbill007** [comment]
Zip is not a streaming format. It has directory at the end of file, so I should store it somewhere first and then unzip. But I think it is possible.

**Comment by tatundria** [comment]
Maybe add an ''update within app'' option?

**Comment by FlyingBananaTree** [comment]
@duckbill007 What do you think about allowing us to configure DBI's background color in `dbi.config`? Would be quite useful to have a black background in applet mode on an OLED switch and maybe some other colors could be added for fun.

**Comment by duckbill007** [comment]
Blue/Black backgrounds were used intentionally to see on screenshot in what mode app is launched. DBI should not be run in title override mode and some its featurtes can not work or work incorrect in this mode.

**Comment by FlyingBananaTree** [comment]
> Blue/Black backgrounds were used intentionally to see on screenshot in what mode app is launched. DBI should not be run in title override mode and some its featurtes can not work or work incorrect in this mode.

Yes you explained that before and I even added it the README (although I wonder which features these are as most of the time I used override mode without any issues).

I think it would be fine if a comment was added in the config file to make sure the user is aware? Maybe you'll consider it. Something like this as an example:

```
; Configure DBI background color, *NOTE* Application mode is not supported and Applet mode should be preferred
AppletBGcolor=green
ApplicationBGcolor=red
```

**Comment by FlyingBananaTree** [comment]
@duckbill007 Just a small thing I noticed, mounting the contents of a title over MTP doesn't respect the automatic screen off setting:

```
; Turn screen off when MTP mode is activated
TurnOffScreen=true
```

The minus button does still work for screen on/off though.
